src/code/views/image-browser-view.coffee

```diff
@@ -0,0 +1,172 @@
+      if @state.selectedImage
+        (PreviewImage {imageInfo: @state.selectedImage, addImage: @addImage})
+      else
+        (div {},
+          (div {className: 'image-browser-form'},
+            (input {ref: 'search', placeholder: 'Search Internal Library and Openclipart.org', value: @state.query, onChange: @changed})
+            (button {}, 'Search')
+          ),
+
+          if showNoResultsAlert
+            (div {className: 'modal-dialog-alert'}, 'Sorry, no images found.  Try another search, or browse internal library images below.')
+
+          (div {className: 'image-browser-header'}, 'Internal Library Images'),
+          (div {className: 'image-browser-results'},
+            if @state.internalResults.length is 0 and (@state.searching or @state.externalResults.length > 0)
+              " No internal library results found for '#{@state.query}'"
+            else
+              for node, index in (if showNoResultsAlert then @state.internalLibrary else @state.internalResults)
+                if node.image and not node.image.match /^(https?|data):/
+                  (ImageSearchResult {key: index, imageInfo: node, clicked: @imageClicked}) if node.image
+          )
+
+          if @state.searchable and not showNoResultsAlert
+            (div {},
+              (div {className: 'image-browser-header'}, 'Openclipart.org Images'),
+              (div {className: 'image-browser-results'},
+                if @state.searching
+                  (div {},
+                    (i {className: "fa fa-cog fa-spin"})
+                    " Searching for #{if @state.searchingAll then 'all matches for ' else ''}'#{@state.query}'..."
+                  )
+                else if @state.externalResults.length is 0
+                  " No openclipart.org results found for '#{@state.query}'"
+                else
+                  for node, index in @state.externalResults
+                    (ImageSearchResult {key: index, imageInfo: node, clicked: @imageClicked})
+              )
+              if @state.externalResults.length < @state.numExternalMatches
+                (div {},
+                  "Showing #{@state.externalResults.length} of #{@state.numExternalMatches} matches for '#{@state.query}'. "
+                  (a {href: '#', onClick: @showAllMatches}, 'Show all matches.')
+                )
+            )
+        )
+    )
+
+MyComputer = React.createFactory React.createClass
+  displayName: 'MyComputer'
+  render: ->
+    (div {}, 'My Computer: TBD')
+
+Link = React.createFactory React.createClass
+  displayName: 'Link'
+  render: ->
+    (div {}, 'Link: TBD')
+
+module.exports = React.createClass
+  displayName: 'Image Browser'
+  render: ->
+    (ModalTabbedDialogFactory {title: (tr "~ADD-NEW-IMAGE.TITLE"), close: @props.close, tabs: [
+      ModalTabbedDialog.Tab {label: (tr "~ADD-NEW-IMAGE.IMAGE-SEARCH-TAB"), component: (ImageSearch {protoNodes: @props.protoNodes, addToPalette: @props.addToPalette})}
```

Comment:
You can see the use of the Tab static function here to create an object to describe a tab.